### DISCUSSION TITLE
Hack for ignoring __generated__ stopped working in 9.5

### DISCUSSION THREAD

**Discussion by TrySound** [vercel/next.js#15565] [Help]
# Bug report

## Describe the bug

We used this hack in app with relay until 9.5
https://github.com/vercel/next.js/issues/8617#issuecomment-527980479

next.js 9.5 started to fail on __generated__ folder again.

Using artifactDirectory is a bad option because will require to rewrite hundreds of modules which import types from __generated__ folders.

## To Reproduce

Disable artifactDirectory in relay examples

## Expected behavior

Some "ignore" option or at least working hack if you don't want to enable it officially.

## Screenshots

```
> Build error occurred
Error: Cannot find module for page: /__generated__/AppQuery.graphql
    at pageNotFoundError (/.../node_modules/next/dist/next-server/server/require.js:1:336)
    at getPagePath (/.../node_modules/next/dist/next-server/server/require.js:2:82)
    at /.../node_modules/next/dist/build/index.js:15:3189
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Promise.all (index 0)
    at async build (/.../node_modules/next/dist/build/index.js:15:2725) {
  code: 'ENOENT'
```

## System information

- OS: macOS
- Version of Next.js: 9.5
- Version of Node.js: 13.12

**Comment by dminkovsky** [comment]
You can configure where relay puts __generated__ files. It's in the docs for the compiler I think, so this hack isn't necessary?

**Comment by TrySound** [comment]
> Using artifactDirectory is a bad option because will require to rewrite hundreds of modules which import types from generated folders.

**Comment by dminkovsky** [comment]
My bad, sorry I missed that. Might be the way, though, and maybe a job for jscodeshift? Something like https://github.com/apollographql/apollo-client/pull/6486/files would make it not too bad and reliable.

**Comment by TrySound** [comment]
I don't think setuping absolute paths aliases for every tool in every project is a good option. Enabling artifactDirectory will make rebase painful for big team working on many features.

**Comment by lgenzelis** [comment]
I solved this issue by adding 

```js
webpack: (config, { webpack }) => {
  config.plugins.push(new webpack.IgnorePlugin(/\/__generated__\//))
  return config
}
```

to `module.exports` in `next.config.js`.
